cpp-ch/local-engine/Parser/SerializedPlanParser.cpp

```diff
@@ -458,9 +475,12 @@ QueryPlanPtr SerializedPlanParser::parseOp(const substrait::Rel & rel, std::list
         }
         case substrait::Rel::RelTypeCase::kRead: {
             const auto & read = rel.read();
-            assert(read.has_local_files() || read.has_extension_table() && "Only support local parquet files or merge tree read rel");
-            if (read.has_local_files())
+            // TODO: We still maintain the old logic of parsing LocalFiles or ExtensionTable in RealRel
+            // to be compatiable with some suites about metrics.
+            // Remove this compatiability in later and then only java iter has local files in ReadRel.
+            if (read.has_local_files() || (!read.has_extension_table() && !isReadFromMergeTree(read)))
```

Comment:
don't add this parameter `isMergeTree`, can ignore the suites about metrics first and raist a issue to fix.

Reply:
the paramter `isMergeTree` is confused.

Reply:
> don't add this parameter `isMergeTree`, can ignore the suites about metrics first and raist a issue to fix.

We need a way to identify whether to parse according to `LocalFiles` or `ExtensionTable` here. If we don't have `isMergeTree` parameter, are there alternative ways to achieve the same thing?

Reply:
`read.has_extension_table()` is for mergetree, and `read.has_local_files()` is for other file format, parquet or orc don't need the extension table part.

Reply:
`ReadRel` no longer includes `LocalFiles` or `ExtensionTable` since they have been decoupled from `RealRel` which is exactly the goal this pr want to achieve, right?

Reply:
Got, can we judge the `split_info` is `localfiles` or `extendsiontable` according to the `split_info string`?

Reply:
I tried before, but unfortunately, there wasn't a suitable way to judge it solely based on `split_info` string information.